text_classification.py

```diff
@@ -0,0 +1,487 @@
+import os
+import re
+import torch
+from torchtext.utils import download_from_url, extract_archive
+from torchtext.data.utils import generate_ngrams
+from torchtext.vocab import build_vocab
+import random
+from torchtext import data
+from tqdm import tqdm
+from torchtext.data import dataset
+from torchtext.data.iterator import generate_iterators
+
+
+class TextDataset(torch.utils.data.Dataset):
+    def __init__(self, examples, fields):
+        self.examples = examples
+        self.fields = fields
+
+    def __getitem__(self, i):
+        return self.examples[i]
+
+    def __len__(self):
+        try:
+            return len(self.examples)
+        except TypeError:
+            return 2**32
```

Comment:
Does this mean self.examples could be something else than a list? Also, why call it examples, isn't it the entire corpus?

Reply:
This is the data loaded from train/test files. It covers the entire corpus in this case.